test/lib/writers/RulesWriter.test.ts

```diff
@@ -0,0 +1,70 @@
+import { OutputFormat } from '@salesforce/code-analyzer-core';
+import fs from 'node:fs';
+import path from 'node:path';
+import { CompositeRulesWriter, RulesFileWriter } from "../../../src/lib/writers/RulesWriter";
+import * as Stub from '../../stubs/StubRuleSelection';
+
+describe('RulesWriter', () => {
+
+    let writeFileSpy: jest.SpyInstance;
+    let writeFileInvocations: { file: fs.PathOrFileDescriptor, contents: string | ArrayBufferView }[];
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+        writeFileInvocations = [];
+        writeFileSpy = jest.spyOn(fs, 'writeFileSync').mockImplementation((file, contents) => {
+            writeFileInvocations.push({file, contents});
+        });
+    });
+
+    describe('RulesFileWriter', () => {
```

Comment:
I would like to see a "Writes file even when no rules are returned" test case like `ResultsWriter.test.ts` has, since that behavior seems desirable to me.